wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -0,0 +1,531 @@
+
+```prolog
+git checkout master
+git branch -d <branchname>
+git push origin :<branchname>
+```
+
+- The first command set the branch to master.
+  This will allow you to delete the other branches.
+- The second command will tell Git to delete the branch.
+  If Git refuse to delete it for a reason (usually the branch was not updated) but you do not want the branch anyway, use the force delete variant by replacing the soft local delete attribute, ``-d``, to force local delete attribute, ``-D``.
+- The third command will update the forked remote repo to also remove the branch (make sure you are connected to the Internet).
+  The semicolon, ``:``, will tells Git to delete this ``<branchname>`` branch in the ``origin`` nickname (which is your forked remote repo).
+
+#### Branch sync
+
+Do a checkout on the branch you want to sync.
+
+```prolog
+git checkout <branchname>
+```
+
+<!-- I do not know what happens in the event of merge conflicts. I usually do it the web way and pull from origin once solved -->
+There are two ways of updating it. The only difference is that in the event of merge conflicts, the first way will catch it and ask you to solve it before continue the merging, while the second way may force you to use _vim_ to manually solve the merge conflicts on the spot to continue the merge. If there is no merge conflicts, both gives the same outcomes.
+
+The first way:
+
+```prolog
+git fetch https://github.com/ppy/osu-wiki.git
+git merge origin/<branchname>
+git push origin <branchname>
+```
+
+The second way:
+
+```prolog
+git pull https://github.com/ppy/osu-wiki.git master
+git push origin <branchname>
+```
+
+_Note: If you have nicknamed https://github.com/ppy/osu-wiki.git, then you can use the nickname instead._
+
+## FAQ
+
+### Why is the file not uploaded?
+
+Currently, 1MB size is the upper limit for any files.
+Resize, or compress the file to stay below the limit.
+
+Also, check the file extension in use.
+Use only lowercased variant of the file extension (like ``.jpg`` instead of ``.JPG``).
+
+### What are the green "Verified" button beside the commits in a PR page?
+
+GitHub supports GPG signature verification as written in their blog post: https://github.com/blog/2144-gpg-signature-verification
+
+It is a personal proof that the commit made by the account user personally, not from a malicious account using the same name.
+By default, all accounts will not have the "Verified" button, as it is an optional commit integrity feature.
+
+To get the "Verified" button, it will take a while, from downloading a tool, generation of key from the tool and some careful configuration.
+
+- For a Windows guide, check this blog post: http://mrexodia.cf/guide/2016/05/30/Github-Gpg
+- For a GitHub's guide, check the GitHub index page for GPG: https://help.github.com/articles/signing-commits-with-gpg/
+
+Do note that it is only applied to Git users only, and is not related to two-factor authorisation.
+
+### I got a merge conflict! Help!
+
```

Comment:
L500: this can be useful for you: [How to Solve Git Conflicts](https://githowto.com/resolving_conflicts)